### DISCUSSION TITLE
error: no member named 'setfill' in namespace 'std'

### DISCUSSION THREAD

**Discussion by yurivict** [facebook/hermes#1484] [General]
## Bug Description

```
/usr/ports/lang/hermes/work/hermes-0.13.0/unittests/BCGen/BytecodeFileFormatTest.cpp:53:16: error: no member named 'setfill' in namespace 'std'
   53 |     ss << std::setfill('0') << std::setw(2) << std::hex
      |           ~~~~~^
/usr/ports/lang/hermes/work/hermes-0.13.0/unittests/BCGen/BytecodeFileFormatTest.cpp:53:37: error: no member named 'setw' in namespace 'std'
   53 |     ss << std::setfill('0') << std::setw(2) << std::hex
      |                                ~~~~~^
2 errors generated.
```

Version: 0.13.0
clang-18
FreeBSD 14.1

**Comment by avp** [comment]
Closing this for now because we can't provide first party support for FreeBSD. We'd be happy to merge any community PRs that fix compilation errors on it though.

**Comment by yurivict** [comment]
@avp 
Is there a particular reason that you've concluded that this is FreeBSD specific?

**Comment by tmikov** [comment]
@yurivict Thank you for your interest in Hermes and for taking the time to report this issue. We appreciate the enthusiasm and effort that goes into building the project in diverse environments.

However, I'd like to clarify our approach to issue submissions. GitHub is a platform where developers collaborate on open-source projects, sharing their work with the community. While we strive to support a broad range of environments, it's important to note that we can't practically investigate and address issues related to environments we haven't tested or configured. You can see the supported and tested environments here:
- https://github.com/facebook/hermes/blob/751d19feb44f6fba04707ffff9111af37dbde220/.circleci/config.yml
- https://github.com/facebook/hermes/blob/751d19feb44f6fba04707ffff9111af37dbde220/.github/workflows/build.yml

In cases like yours, where the error occurs in a non-standard or untested environment (such as FreeBSD with Clang 18), we encourage the community to discuss and troubleshoot these challenges together in the Discussions section. This allows for a more collaborative approach, where others who might have encountered similar issues can offer insights or solutions.

If you manage to identify the problem and have a fix, we would be very grateful if you could submit a pull request (PR). This kind of peer collaboration is what makes open source thrive.

Thank you again for your understanding and for contributing to the Hermes project.
